### ISSUE TITLE
Problem with pull images from registry.redhat.io to OpenShift 3.11.

### ISSUE THREAD

**Issue by phracek** [openshift/origin#24539]
We, RHSCL team, would like to test our containers with registry.redhat.io

Unfortunately, we got error like:
```
++ echo 'INFO: printing logs for pod ruby-testing-1-build'
++ oc logs ruby-testing-1-build
pulling image error: Get https://registry.redhat.io/v2/rhscl/ruby-24-rhel7/manifests/sha256:b4221849f931c7f34255638902c775d17a1a009970f7b49f63366f0ab2e33a7f: unauthorized: Please login to the Red Hat Registry using your Customer Portal credentials. Further instructions can be found here: https://access.redhat.com/RegistryAuthentication
error: build error: unable to get registry.redhat.io/rhscl/ruby-24-rhel7@sha256:b4221849f931c7f34255638902c775d17a1a009970f7b49f63366f0ab2e33a7f
OpenShift tests for rhscl/ruby-24-rhel7 failed.
```
The full log is here: https://gist.github.com/rhscl-bot/e06cc09e7fa9105f4c42222220658b13#file-s2i-ruby-container-29-sh-L6738

##### Version
**https://gist.github.com/rhscl-bot/e06cc09e7fa9105f4c42222220658b13#file-s2i-ruby-container-29-sh-L4209**

```
oc v3.11.154
kubernetes v1.11.0+d4cacc0
features: Basic-Auth GSSAPI Kerberos SPNEGO
Server https://127.0.0.1:8443
kubernetes v1.11.0+d4cacc0
```

##### Steps To Reproduce
1. [step 1]
2. [step 2]

##### Current Result
We are not able tu pull images from registry.redhat.io.
##### Expected Result
We are able tu pull images from registry.redhat.io.

##### Additional Information
Alone tests start here: https://gist.github.com/rhscl-bot/e06cc09e7fa9105f4c42222220658b13#file-s2i-ruby-container-29-sh-L4060

Starting OpenShift cluster is here: https://gist.github.com/rhscl-bot/e06cc09e7fa9105f4c42222220658b13#file-s2i-ruby-container-29-sh-L4108
OpenShift server was started properly.

OpenShift version is here: https://gist.github.com/rhscl-bot/e06cc09e7fa9105f4c42222220658b13#file-s2i-ruby-container-29-sh-L4209

Alone test starts here: https://gist.github.com/rhscl-bot/e06cc09e7fa9105f4c42222220658b13#file-s2i-ruby-container-29-sh-L4917

Adding secret is here: https://gist.github.com/rhscl-bot/e06cc09e7fa9105f4c42222220658b13#file-s2i-ruby-container-29-sh-L4926
JSON format is here: https://gist.github.com/rhscl-bot/e06cc09e7fa9105f4c42222220658b13#file-s2i-ruby-container-29-sh-L4932

Login to registry.redhat.io registry by `docker login -p <OUR_TOKEN> -u='6340056|core-services-apps-ci' registry.redhat.io` works properly.
But we are not able to do the same thing with OpenShift 3.11.
What can be a problem?

Adding to CC @pkubat and @hhorak Since I will be on vacation since 23rd February.

**Comment by adambkaplan** [comment]
Pulling images directly from `registry.redhat.io` requires a pull secret. See the docs [1].

[1] https://docs.okd.io/3.11/dev_guide/builds/build_inputs.html#using-docker-credentials-for-private-registries

**Comment by pkubatrh** [comment]
>Pulling images directly from registry.redhat.io requires a pull secret.

Yes, hence the addition of the secret is explicitly mentioned in the report.

We have also tried to add the pull secret to the openshift namespace, but without any success.

**Comment by pkubatrh** [comment]
@adambkaplan 

Managed to narrow the issue down to a combination of `oc import-image` only importing image metadata and not the image itself, and the build pod not being able to access the registry for some reason.

I saw that buildConfig has a way to define a pullSecret so I tried that and it worked like a charm. Is there any other way to get to the same conclusion without having to patch every single buildConfig that gets created?
Alternatively, is there any way to force an image (content) sync while importing an image? I expect the build pod would have no issues if the image was in the local registries already...
